scheduler/pkg/store/utils/validate.go

```diff
@@ -0,0 +1,11 @@
+package utils
+
+import "regexp"
+
+func CheckName(name string) bool {
+	ok, err := regexp.Match("^[a-zA-Z0-9][a-zA-Z0-9-_]*[a-zA-Z0-9]$", []byte(name))
```

Comment:
The spec on what names are allowed in k8s is this one: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

k8s will already reject all names that don't obey the regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*` (this is the regex that kubectl reports when trying to create a resource with an invalid name). In particular, upper-case letters and underscores are also excluded.

We'll still want to do this validation explicitly so that we deal in an uniform way with k8s and non-k8s deployments.

Beyond k8s's existing validation, we want to reserve the `.` for Seldon -- this is because internally we do header-based routing and a pipeline's name (for example) is referred to as '[name].pipeline`. 

I'm not sure why we don't just trim the element after the _last_ dot when post-processing such names (and allow dots in user-provided names). I don't have the context on whether there are any other reasons why we might reserve the `.`, but perhaps we take this as a separate issue to investigate.

Reply:
Yes, some more investigation is necessary. It would be great to have this validation aligned with k8s.